src/middlewared/middlewared/plugins/service_/services/kubernetes.py

```diff
@@ -58,6 +62,23 @@ async def before_start(self):
         await self.middleware.call('service.start', 'docker')
         await self._systemd_unit('cni-dhcp', 'start')
 
+        # It is possible server/token got misconfigured somehow ( we have had reports that on unclean shutdown the file
+        # is empty or malformed, in this case we should remove the token as it's generated automatically by k3s again
+        # and it's not useful to us as well per se as it's used for multi-node cluster )
+
+        await self.middleware.run_in_thread(
+            self.check_server_token, (await self.middleware.call('kubernetes.config'))['dataset']
+        )
+
+    def check_server_token(self, k8s_dataset):
+        token_path = os.path.join('/mnt', k8s_dataset, 'k3s/server/token')
+        if os.path.exists(token_path):
```

Comment:
Checking a file for its' existence before opening it is prone to a race condition. I'd do something like this
```
token = ''
with contextlib.suppress(FileNotFoundError):
    with open(token_path) as f:
        token = f.read().strip()

if not RE_TOKEN_FORMAT.findall(token):
    os.unlink(token_path)
```